vortex-duckdb-ext/build.rs

```diff
@@ -0,0 +1,59 @@
+use std::env;
+use std::path::PathBuf;
+
+use walkdir::WalkDir;
+
+fn main() {
+    let crate_dir = PathBuf::from(env::var("CARGO_MANIFEST_DIR").unwrap());
+    // Directory of our DuckDB extension.
+    let duckdb_ext_dir = crate_dir.parent().unwrap().join("duckdb-vortex");
+
+    // Generate the _imported_ bindings from our C++ code.
+    bindgen::Builder::default()
+        .header("cpp/include/duckdb_vx.h")
+        // Add the #[must_use] attribute to FFI functions that return results.
+        .must_use_type("duckdb_state")
+        .rustified_enum("duckdb_state")
+        .rustified_non_exhaustive_enum("DUCKDB_TYPE")
+        .size_t_is_usize(true)
+        //.generate_cstr(true) // This emits invalid syntax and breaks the Rust formatter
+        .clang_arg(format!(
+            "-I{}",
+            duckdb_ext_dir.join("duckdb/src/include/").to_str().unwrap()
+        ))
+        // Tell cargo to invalidate the built crate whenever any of the
+        // included header files changed.
+        .parse_callbacks(Box::new(bindgen::CargoCallbacks::new()))
+        // Finish the builder and generate the bindings.
+        .generate()
+        // Unwrap the Result and panic on failure.
+        .expect("Unable to generate bindings")
+        .write_to_file(crate_dir.join("src/cpp.rs"))
+        .expect("Couldn't write bindings!");
+
+    // Compile our C++ code that exposes additional DuckDB functionality.
+    cc::Build::new()
+        .std("c++17")
+        // Allow C++20 designator syntax even with C++17 std
+        .flag("-Wno-c++20-designator")
```

Comment:
Given that DuckDB dropped the `gcc4` target as part of their community extension CI, I think we can simply go C++ 20 now, no exception for `-Wno-c++20-designator` needed.